sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -261,10 +271,37 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession {
       }
     }.flatten
 
+    val tempQueries = if (code.exists(_.trim.startsWith("--QUERY-DELIMITER"))) {
+      // Although the loop is heavy, only used for comments test.
+      val querys = new ArrayBuffer[String]
+      val otherCodes = new ArrayBuffer[String]
+      var tempStr = ""
+      var start = false
+      for (c <- code) {
+        if (c.trim.startsWith("--QUERY-DELIMITER-START")) {
+          start = true
+          querys ++= otherCodes.toSeq.mkString("\n").split("(?<=[^\\\\]);")
+          otherCodes.clear()
+        } else if (c.trim.startsWith("--QUERY-DELIMITER-END")) {
+          start = false
+          if (tempStr.endsWith(";")) {
+            tempStr = tempStr.substring(0, tempStr.length - 1)
+          }
+          querys += s"\n$tempStr"
```

Comment:
this can be `querys += s"\n${tempStr.stripSuffix(";")}"`

Reply:
OK. Good idea.